File: sigv4-signing/rust/src/sigv4_signer_constants.rs
pub mod constants {
    pub const ALGORITHM_AWS4_HMAC_SHA_256: &str = "AWS4-HMAC-SHA256";
    pub const AWS4_REQUEST_TYPE: &str = "aws4_request";
    pub const SERVICE: &str = "kinesisvideo";
    pub const X_AMZ_ALGORITHM: &str = "X-Amz-Algorithm";
    pub const X_AMZ_CREDENTIAL: &str = "X-Amz-Credential";
    pub const X_AMZ_DATE: &str = "X-Amz-Date";
    pub const X_AMZ_EXPIRES: &str = "X-Amz-Expires";
    pub const X_AMZ_SECURITY_TOKEN: &str = "X-Amz-Security-Token";
    pub const X_AMZ_SIGNATURE: &str = "X-Amz-Signature";
    pub const X_AMZ_SIGNED_HEADERS: &str = "X-Amz-SignedHeaders";
    pub const NEW_LINE_DELIMITER: &str = "\n";
    pub const DATE_PATTERN: &str = "%Y%m%d";
    pub const TIME_PATTERN: &str = "%Y%m%dT%H%M%SZ";
    pub const METHOD: &str = "GET";
    pub const SIGNED_HEADERS: &str = "host";
}

File: sigv4-signing/go/pkg/sigv4signer/sigv4signer_constants.go
package sigv4signer

const (
	ALGORITHM_AWS4_HMAC_SHA_256 = "AWS4-HMAC-SHA256"
	AWS4_REQUEST_TYPE           = "aws4_request"
	SERVICE                     = "kinesisvideo"
	X_AMZ_ALGORITHM             = "X-Amz-Algorithm"
	X_AMZ_CREDENTIAL            = "X-Amz-Credential"
	X_AMZ_DATE                  = "X-Amz-Date"
	X_AMZ_EXPIRES               = "X-Amz-Expires"
	X_AMZ_SECURITY_TOKEN        = "X-Amz-Security-Token"
	X_AMZ_SIGNATURE             = "X-Amz-Signature"
	X_AMZ_SIGNED_HEADERS        = "X-Amz-SignedHeaders"
	NEW_LINE_DELIMITER          = "\n"
	DATE_PATTERN                = "20060102"
	TIME_PATTERN                = "20060102T150405Z"
	METHOD                      = "GET"
	SIGNED_HEADERS              = "host"
)
